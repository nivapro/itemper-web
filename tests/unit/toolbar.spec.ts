
import Vue from 'vue';
import Toolbar from '../../src/components/toolbar.vue';

import Vuetify from 'vuetify';

import { createLocalVue, mount } from '@vue/test-utils';

// import { shallow } from '@vue/test-utils';

describe('toolbar.spec.ts', () => {
  let wrp: any;
  let vm: any;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuetify);

    wrp = mount(Toolbar, {
      localVue,
    });
    vm = wrp;
  });

  it('does not display element if there are no items', () => {
    // No items to display.
    // NOTE: By default, our component has some items, but we use
    // setData() here to override it with an empty array of items to
    // make sure we have not items at all and our test works as
    // expected.
    wrp.setData({ menuItems: [] });
    expect(wrp.find('.item-index-row').exists()).toBe(false);

  });

  it('has the expected html structure', () => {
    expect(wrp.element).toMatchSnapshot();
  });

  it('has default menu items ', () => {
    expect(wrp.vm.menuItems[3].route).toBeDefined();
  });
});
