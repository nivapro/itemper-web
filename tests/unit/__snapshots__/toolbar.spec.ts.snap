// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toolbar.spec.ts has the expected html structure 1`] = `
<nav
  class="toolbar theme--dark light-blue"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px;"
>
  <div
    class="toolbar__content"
    style="height: 56px;"
  >
    <div
      class="menu"
      style="display: inline-block;"
    >
      <div
        class="menu__activator"
      >
        <button
          class="btn btn--icon"
          type="button"
        >
          <div
            class="btn__content"
          >
            <button
              class="toolbar__side-icon btn btn--icon"
              type="button"
            >
              <div
                class="btn__content"
              >
                <i
                  aria-hidden="true"
                  class="icon material-icons"
                >
                  menu
                </i>
              </div>
            </button>
          </div>
        </button>
      </div>
      <div
        class="menu__content"
        style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; z-index: 0; display: none;"
      >
         
        <li
          class="subheader"
        />
         
        <ul
          class="list"
        >
          <li>
            <a
              class="list__tile list__tile--link"
            >
              <div
                class="list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="icon blue--text text--darken-2 material-icons"
                >
                  home
                </i>
              </div>
               
              <div
                class="list__tile__title"
              >
                Hem
              </div>
            </a>
          </li>
          <li>
            <a
              class="list__tile list__tile--link"
            >
              <div
                class="list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="icon blue-grey--text text--darken-2 material-icons"
                >
                  place
                </i>
              </div>
               
              <div
                class="list__tile__title"
              >
                Platser
              </div>
            </a>
          </li>
          <li>
            <a
              class="list__tile list__tile--link"
            >
              <div
                class="list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="icon blue-grey--text text--darken-2 material-icons"
                >
                  poll
                </i>
              </div>
               
              <div
                class="list__tile__title"
              >
                Givare
              </div>
            </a>
          </li>
          <li>
            <a
              class="list__tile list__tile--link"
            >
              <div
                class="list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="icon blue-grey--text text--darken-2 material-icons"
                >
                  settings
                </i>
              </div>
               
              <div
                class="list__tile__title"
              >
                Inst√§llningar
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="toolbar__title"
    >
      iTemper
    </div>
     
    <div
      class="spacer"
    />
     
    <button
      class="btn btn--icon"
      type="button"
    >
      <div
        class="btn__content"
      >
        <i
          aria-hidden="true"
          class="icon material-icons"
        >
          search
        </i>
      </div>
    </button>
     
    <button
      class="btn btn--icon"
      type="button"
    >
      <div
        class="btn__content"
      >
        <i
          aria-hidden="true"
          class="icon material-icons"
        >
          view_module
        </i>
      </div>
    </button>
  </div>
</nav>
`;
